### PR TITLE
add performance demo mode

### PR DESCRIPTION
**Pull request by fergald**
Add a button that enables an FPS display and animates resizing the contact list horizontally. Between virtual-scroller and raw divs demonstrates clear drop in FPS and increase in CPU.

### CODE REVIEW

File: demo/sticky.html [hunk 1]
```diff
@@ -54,12 +68,16 @@
 
   <div>
     <button id=swapButton>switch virtual-scroller &lt;-&gt; div</button>
+    <button id=perfDemoModeButton>performance demo mode</button>
   </div>
 
-  <virtual-scroller id=scroller></virtual-scroller>
-
+  <fps-display style="display: none"></fps-display>
```
Comment (domenic) [inline demo/sticky.html]: Personally I would use the `hidden` attribute (plus `toggleAttribute("hidden", perfModeDemoState)`)

File: demo/sticky.html [hunk 2]
```diff
@@ -131,6 +149,32 @@ <h1>${firstLetter}</h1>
...
+    let perfDemoModeState = false;
+
+    /**
+     * Swaps between performance demo mode and not. In this mode we
+     * should the FPS and we do an animated resize of the contacts list.
+     */
+    function perfDemoMode() {
+      let fpsDisplayStyleDisplay;
+      let scrollContainerClass;
+      if (perfDemoModeState === false) {
```
Comment (domenic) [inline demo/sticky.html]: Do `if (perfDemoModeState) { ... } else { ... }`

File: demo/sticky.html [hunk 3]
```diff
@@ -131,6 +149,32 @@ <h1>${firstLetter}</h1>
     }
     document.querySelector('#swapButton')
         .addEventListener('click', swapScrollers);
+
+    let perfDemoModeState = false;
```
Comment (domenic) [inline demo/sticky.html]: Maybe `isInPerfDemoMode`

**Comment by domenic** [comment]
I tried this on mobile and you can't see the resize. (However FPS still drop drastically with divs.) It might be worth tweaking the stylesheet a bit to make that work.

File: demo/sticky.html (hunk 2 above)
Comment (fergald) [reply demo/sticky.html]: yeah, no idea what I was doing there. Anyway the code is a lot simpler now.

File: demo/sticky.html (hunk 3 above)
Comment (fergald) [reply demo/sticky.html]: done.

File: demo/sticky.html (hunk 1 above)
Comment (fergald) [reply demo/sticky.html]: I started to do that but then that either gives me 2 indepdent bits for the state of whether we're in demo mode or not (one in JS, one in this attribute) or I can switch to make this the canonical bit but that feels weird.

But I did switch to toggling the hidden attribute.

**Comment by fergald** [comment]
I'm going to submit this and look into fixing mobile after.

**Pull request merged**